service/controller/v3/resource/vpngateway/update_test.go

```diff
@@ -0,0 +1,250 @@
+package vpngateway
+
+import (
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2017-09-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+func TestNeedUpdate(t *testing.T) {
+	testCases := []struct {
+		description      string
+		current, desired network.VirtualNetworkGatewayConnection
+		expected         bool
+	}{
+		{
+			"desired state empty",
```

Comment:
Please just initialize the struct with the keys being visible. This is by no means useful. Lets not be lazy on the wrong place. You cannot easily maintain nor extend this. You cannot read it. Tomorrow somebody else has to work with this and has no idea what is going on and when the order of the struct properties changes all this here is immediately fucked and we all just waste a lot of time.

Reply:
done